server/src/internalClusterTest/java/org/opensearch/snapshots/SearchableSnapshotIT.java

```diff
@@ -547,24 +554,58 @@ public void testCacheFilesAreClosedAfterUse() throws Exception {
         final Client client = client();
 
         internalCluster().ensureAtLeastNumSearchAndDataNodes(numReplicasIndex + 1);
-        createIndexWithDocsAndEnsureGreen(1, 100, indexName);
+        createIndexWithDocsAndEnsureGreen(numReplicasIndex, 100, indexName);
         createRepositoryWithSettings(null, repoName);
         takeSnapshot(client, snapshotName, repoName, indexName);
         restoreSnapshotAndEnsureGreen(client, snapshotName, repoName);
         assertDocCount(restoredIndexName, 100L);
 
+        // Get the available NodeEnvironment instances
+        Iterable<NodeEnvironment> nodeEnvironments = internalCluster().getInstances(NodeEnvironment.class);
+
+        // Filter out search NodeEnvironment(s) since FileCache is initialized only on search nodes and
+        // collect the path for all the cache locations on search nodes.
+        List<Path> searchNodeFileCachePaths = StreamSupport.stream(nodeEnvironments.spliterator(), false)
+            .filter(nodeEnv -> nodeEnv.fileCache() != null)
+            .map(nodeEnv -> nodeEnv.fileCacheNodePath().fileCachePath)
+            .collect(Collectors.toList());
+
+        for (Path fileCachePath : searchNodeFileCachePaths) {
+            assertTrue(Files.exists(fileCachePath));
+            assertTrue(Files.isDirectory(fileCachePath));
+            try (DirectoryStream<Path> pathStream = Files.newDirectoryStream(fileCachePath)) {
+                for (Path dataPath : pathStream) {
```

Comment:
Do we expect there to only be one here as well? This test would pass if the path stream was empty, I think. Maybe better to do something like `pathStream.iterator().next()` to rely on there being exactly one.

Reply:
That's the subdirectory within the cache folder for the `nodeLockId`. There should be only 1 folder there. 
https://github.com/opensearch-project/OpenSearch/blob/71a76da0d894181e909237f50522c3e475a9540e/server/src/main/java/org/opensearch/env/NodeEnvironment.java#L1359-L1363